tests/unit/mesh/mesh_test_utils.hpp

```diff
@@ -0,0 +1,215 @@
+ * condition.
+ */
+void CheckPoisson(ParMesh &pmesh, int order,
+                  int disabled_boundary_attribute = -1);
+
+/**
+ * @brief Check that a ParMesh generates the same number of boundary elements as
+ * the serial mesh.
+ *
+ * @param smesh Serial mesh to be built from and compared against
+ * @param partition Optional partition
+ * @return std::unique_ptr<ParMesh> Pointer to the mesh in question.
+ */
+std::unique_ptr<ParMesh> CheckParMeshNBE(Mesh &smesh,
+                                         const std::unique_ptr<int[]> &partition = nullptr);
+
+/**
+ * @brief Helper function to track if a face index is internal
+ *
+ * @param pmesh The mesh containing the face
+ * @param f The face index
+ * @param local_to_shared A map from local faces to shared faces
+ * @return true the face is between domain attributes (and owned by this rank)
+ * @return false the face is not between domain attributes or not owned by this
+ * rank
+ */
+bool CheckFaceInternal(ParMesh& pmesh, int f,
+                       const std::map<int, int> &local_to_shared);
+
+/**
+ * @brief Given a parallel and a serial mesh, perform an L2 projection and check
+ * the solutions match exactly.
+ * @details The result of an L2 projection on a parallel mesh and serial mesh
+ * should be identical irrespective of partitioning. This check can be used for
+ * generating data for comparison of a parallel and serial mesh.
+ *
+ * @param pmesh The parallel mesh to use in the AB test
+ * @param smesh The serial mesh to use in the AB test
+ * @param order The polynomial order to use in the projection
+ * @param exact_soln A function that returns the exact solution at a given point
+ * @return std::array<double, 2> Pair of error on the serial mesh and the
+ * parallel mesh. Should be within numerical tolerance of each other.
+ */
+std::array<double, 2> CheckL2Projection(ParMesh& pmesh, Mesh& smesh, int order,
+                                        std::function<double(Vector const&)> exact_soln);
+
+
+/**
+ * @brief Helper for counting the number of essential local degrees of freedom
+ * within a parallel mesh.
+ *
+ * @tparam FECollection FiniteElementCollection to define the space.
+ * @tparam TDOF Whether or not to counter true degrees of freedom (TDOF) or
+ * local (vector) degrees of freedom (VDOF).
+ * @param mesh The mesh to perform the test upon.
+ * @param order The polynomial order of the basis.
+ * @param attribute The attribute of the boundary to count the essential DOFs
+ * on.
+ * @return int The number of essential DOFs.
+ */
+template <typename FECollection, bool TDOF = true>
+int CountEssentialDof(ParMesh &mesh, int order, int attribute);
+
+/**
+ * @brief Helper for counting the number of essential degrees of freedom within
+ * a parallel mesh, and summing over all processors.
+ *
+ * @tparam FECollection FiniteElementCollection to define the space.
+ * @tparam TDOF Whether or not to counter true degrees of freedom (TDOF) or
```

Comment:
... count ...

Reply:
Fixed in https://github.com/mfem/mfem/pull/3746/commits/4fe30263e4beb521564a5a2561501e8bbd67900c